src/unixfs/mod.rs

```diff
@@ -77,4 +177,66 @@ mod tests {
         let cid2 = file.put_unixfs_v1(&dag).await.unwrap();
         assert_eq!(cid.to_string(), cid2.to_string());
     }
+
+    #[ignore]
+    #[async_std::test]
+    async fn cat() {
+        use crate::{IpfsOptions, UninitializedIpfs};
+        use async_std::task;
+        use futures::stream::StreamExt;
+        use hex_literal::hex;
+        use sha2::{Digest, Sha256};
+        use std::time::{Duration, Instant};
+
+        let options = IpfsOptions::inmemory_with_generated_keys(false);
+
+        let (ipfs, fut) = UninitializedIpfs::new(options).await.start().await.unwrap();
+        task::spawn(fut);
+
+        let id = ipfs.identity().await.unwrap();
+        println!("listening at: {}/p2p/{}", id.1[0], id.0.into_peer_id());
+        let cid = "QmTEn8ypAkbJXZUXCRHBorwF2jM8uTUW9yRLzrcQouSoD4";
+        println!("please connect an go-ipfs with Cid {}", cid);
+
+        while ipfs.peers().await.unwrap().is_empty() {
+            async_std::future::timeout(
+                Duration::from_millis(100),
+                futures::future::pending::<()>(),
+            )
+            .await
+            .unwrap_err();
+        }
+
+        println!("got peer");
+        let started_at = Instant::now();
+
+        let stream = super::cat(ipfs, libipld::cid::Cid::try_from(cid).unwrap(), None);
+
+        futures::pin_mut!(stream);
+
+        let mut digest = Sha256::new();
+        let mut count = 0;
+
+        loop {
+            let bytes = match stream.next().await {
```

Comment:
nit: wouldn't `while let Some(x) = stream.next().await`  work here?

Reply:
Yes it does, but clippy did not mention on this. Perhaps because this was a match on `Option<Result<_, _>>`.